core/shared/src/main/scala/zio/prelude/AssociativeEither.scala

```diff
@@ -214,27 +212,22 @@ object AssociativeEither extends LawfulF.Invariant[AssociativeEitherDeriveEqualI
     }
 
   /**
-   * The `AssociativeEither` instance for `ZStream`.
+   * The `CommutativeEither` (and `AssociativeEither`) instance for `ZStream`.
    */
-  implicit def ZStreamAssociativeEither[R, E]: AssociativeEither[({ type lambda[+a] = ZStream[R, E, a] })#lambda] =
-    new AssociativeEither[({ type lambda[+a] = ZStream[R, E, a] })#lambda] {
+  implicit def ZStreamCommutativeEither[R, E]: CommutativeEither[({ type lambda[+a] = ZStream[R, E, a] })#lambda] =
+    new CommutativeEither[({ type lambda[+a] = ZStream[R, E, a] })#lambda] {
       def either[A, B](fa: => ZStream[R, E, A], fb: => ZStream[R, E, B]): ZStream[R, E, Either[A, B]] =
-        fa.map(Left(_)) orElse fb.map(Right(_))
+        fa mergeEither fb
     }
 
   /**
-   * The `AssociativeEither` instance for failed `ZStream`.
+   * The `CommutativeEither` (and `AssociativeEither`) instance for `ZSink`.
    */
-  implicit def ZStreamFailureAssociativeEither[R, A]
```

Comment:
We should keep `ZStreamFailureAssociativeEither`, it's about `Failure[ZStream[R, +*, A]]`, so no conflict with `ZStream[R, E, +*]`.

Reply:
Done!